### ISSUE TITLE
Graph not updating index

### ISSUE THREAD

**Issue by sajid2045** [orientechnologies/orientdb#4907]
On version 2.1.1, 

I create a fresh cluster of 3 nodes,

I run the following script on 1 of the node (via studio):

```sql
CREATE class Topic extends V
CREATE PROPERTY Topic.name STRING
CREATE PROPERTY Topic.created DATETIME


CREATE class Root extends Topic
CREATE INDEX Root.name NOTUNIQUE
CREATE class Event extends Topic
CREATE INDEX Event.name NOTUNIQUE

CREATE class Subscriber extends V
CREATE PROPERTY Subscriber.subscriptionId STRING
CREATE PROPERTY Subscriber.subscriberType STRING
CREATE PROPERTY Subscriber.name STRING
CREATE PROPERTY Subscriber.created DATETIME




CREATE class Device extends Subscriber
CREATE PROPERTY Device.deviceId STRING
CREATE PROPERTY Device.type STRING
CREATE PROPERTY Device.vendor STRING
CREATE INDEX Device.subscriptionId UNIQUE


CREATE class User extends Subscriber
CREATE PROPERTY User.login STRING
CREATE INDEX User.subscriptionId UNIQUE
CREATE INDEX User.login UNIQUE


CREATE class SubscriptionsJson extends V
CREATE PROPERTY SubscriptionsJson.uniqueDocumentId STRING
CREATE INDEX SubscriptionsJson.uniqueDocumentId UNIQUE
```

Now, I create some Entries like Device & User

But when I do `select from User where subscriptionId = 'xxx'`, it does not return anything, but once I rebuild the index, it returns the record. This I need to do everytime I add a new Entity. 

Any pointers?

**Comment by sajid2045** [comment]
https://dl.dropboxusercontent.com/u/5968302/OrientE1.PNG

**Comment by lvca** [comment]
Are you using synchronous replication?

**Comment by sajid2045** [comment]
Yes, 

Also, it looks like a cluster problem, when running in standalone mode, everything works fine!

**Comment by lvca** [comment]
By reading the logs I found something could be the problem. I'll keep you posted soon.

**Comment by sajid2045** [comment]
Hi , 

Although this issue is tagged as Medium, It is a blocker from us running in clustered mode. Can you reproduce this issue on your side? I am just wondering if its a bug or setup / index creation issue on our side. Either way, one would hope the system to behave same in clustered vs standalone mode.

**Comment by lvca** [comment]
This problem could be related to: https://github.com/orientechnologies/orientdb/issues/4913. Working to reproduce it.

**Comment by lvca** [comment]
With synchronous replication I'm not able to reproduce it. Trying with asynch.

**Comment by lvca** [comment]
With synchronous replication I'm not able to reproduce it. Trying with asynch.

**Comment by sajid2045** [comment]
That is strange. What process are you following to add the schema.sql?

I tried creating a fresh 3 node synchronous cluster. 

Then on one of the workbench , I run the sql scripts one by one. Not sure if thats the right way of doing it.

**Comment by lvca** [comment]
Tried with async replication and works. How can I reproduce it? Do you have any logs? Please could you send them to support@ ?

**Comment by sajid2045** [comment]
which log you are after?

**Comment by lvca** [comment]
I started 3 nodes and executed your exact commands to create the schema, the I've create 3 User and query from other nodes by using the index and result was returned.

**Comment by sajid2045** [comment]
This looks like working after I completely removed orientdb instances directory and did a fresh install. I think the DB entered a corrupted state somehow.

What is a proper way of deleting the data directories of orient so it becomes a clean install?

**Comment by lvca** [comment]
Just delete the database directory and OrientDB will ask a fresh copy to the existent server cluster.

**Comment by coldfuzz** [comment]
Hi Luca;
I also get the same problem about index at version 2.1.1
This problem is solved with 2.1.2 or you offer another solution.
Best Regards.
